clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -109,9 +109,9 @@ public Metrics(MetricConfig defaultConfig, List<MetricsReporter> reporters, Time
      */
     public Metrics(MetricConfig defaultConfig, List<MetricsReporter> reporters, Time time, boolean enableExpiration) {
         this.config = defaultConfig;
-        this.sensors = new CopyOnWriteMap<>();
-        this.metrics = new CopyOnWriteMap<>();
-        this.childrenSensors = new CopyOnWriteMap<>();
+        this.sensors = new ConcurrentHashMap<>();
```

Comment:
Yes I was wondering if you had percentile times to insert any given entry. With a c-o-w map, each insert gets progressively worse so over time even the lower percentiles (50th/75th, etc.) have very high insert times. I believe for the concurrent map perhaps only the 99.99999...th percentile would have high latency (due to the need to resize). The only issue with the concurrent map as I wrote earlier is that the specification does not offer any guarantee on its put/get time (all it says is that it is functionally spec is the same as Hashtable). It is a bit unfortunate that we cannot use a c-o-w map (which would otherwise be ideal for metrics because intuitively you typically wouldn’t expect metrics to get a lot of sudden inserts from request handler threads) but quotas changes that. In fact an alternate would be to just use a simple HashMap (which does offer constant-time under most circumstances) and surround all operations on it with a read/write lock as appropriate. Perhaps I’m over-thinking this given that we use concurrent maps all over the place but wondering if you have any preferences/further thoughts given the above.